docs/developer_guide/style_guide.rst

```diff
@@ -0,0 +1,28 @@
+Style Guide and General Conventions
+-----------------------------------
+
+As a general rule we use `black <https://black.readthedocs.io/en/stable/>`_ formater which should take care of most of
+the formating issues. To ensure that your commits are already formatted the repo already contains the machinery
+for per-commit hooks. To enable black formating do the following:
+
+1. Install pre-commit: :code:`pip install pre-commit``
+2. Execute :code:`pre-commit install`` to install git hooks in your .git/ directory.
+
+For what is not covered by black, this project follows some of the conventions to `scikit-learn <https://scikit-learn.org/stable/>`_.
+
+#. Use underscores to separate words in non class names: :code:`n_samples` rather than :code:`nsamples`.
+#. Avoid multiple statements on one line. Prefer a line return after a control flow statement (if/for).
+#. Use relative imports for references inside the source code :code:`src/neuroconv`.
+#. Unit tests are an exception to the previous rule; they should use absolute imports, exactly as client code would.
+#. A corollary is that, if :code:`neuroconv.foo`` exports a class or function that is implemented in :code:`neuroconv.foo.bar.baz`, the test should import it from :code:`neuroconv.foo`.
+#. Please dont use :code:`import *` in any case. It is considered harmful by the official Python recommendations. It makes the code harder to read as the origin of symbols is no longer explicitly referenced, but most important, it prevents using a static analysis tool like pyflakes to automatically find bugs in scikit-learn.
+#. Use the numpy `docstring standard <https://numpydoc.readthedocs.io/en/latest/format.html#numpydoc-docstring-guide>`_ in all your docstring.
+
+Data interfaces conventions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#. Use :code:`file_path` and :code:`folder_path` as arguments for the location of input files and folder respectively.
+#. In opposition to convention number 1 above, we use :code:`nwbfile` to refer to in-memory filesin the nwb format.
```

Comment:
```suggestion
#. In opposition to convention number 1 above, we use :code:`nwbfile` to refer to in-memory files in the nwb format.
```

Reply:
```suggestion
#. Use  :code:`nwbfile` as the variable name to refer to an in-memory :py:class:`~pynwb.NWBFile` object.
```

I wouldn't even say it's an exception either - the `_path` references are to places on disk, the `nwbfile` object in memory may not even have been written to disk at all.

Reply:
This is referring to 

> #. Use underscores to separate words in non class names: :code:`n_samples` rather than :code:`nsamples`.

Reply:
That's a bit unclear due to the rendering of the numbers on readthedocs, then

![image](https://user-images.githubusercontent.com/51133164/180483953-a1588b6c-cb77-4212-8825-a6baea7bd7a4.png)

link: https://neuroconv--28.org.readthedocs.build/en/28/developer_guide/style_guide.html

Reply:
@h-mayorquin Also there's some trailing ` characters for some of those code references